pramen/runner/src/main/scala/za/co/absa/pramen/runner/cmd/CmdLineConfig.scala

```diff
@@ -60,117 +61,74 @@ object CmdLineConfig {
   }
 
   def parseCmdLine(args: Array[String]): Option[CmdLineConfig] = {
-    val parser = new CmdParser("spark-submit pipeline-runner.jar " +
-      "za.co.absa.pramen.core.runner.PipelineRunner " +
-      "--workflow <Workflow Configuration Path> " +
-      "[--files <comma-separated list of files> " +
-      "[--ops output_table1,output_table2,...]" +
-      "[--date <Current date override (yyyy-MM-dd)>] " +
-      "[--rerun <The date to force rerun (yyyy-MM-dd)>] " +
-      "[--verbose] "  +
-      "[--dry-run] " +
-      "[--check-late-only]" +
-      "[--check-new-only] " +
-      "[--undercover] " +
-      "[--use-lock <true/false>] " +
-      "[--date-from <date_from>]" +
-      "[--date-to <date_to>]" +
-      "[--run-mode { fill_gaps | check_updates | force }]" +
-      "[--inverse-order <true/false>]" +
-      "[--override-log-level <log_level>]"
-    )
+    val parser = new CmdParser("spark-submit pipeline-runner.jar za.co.absa.pramen.core.runner.PipelineRunner")
 
     parser.parse(args, CmdLineConfig())
   }
 
   def applyCmdLineToConfig(conf: Config, cmd: CmdLineConfig): Config = {
     val dateFormatter = DateTimeFormatter.ofPattern(Constants.DATE_FORMAT_INTERNAL)
+    var accumulatedConfig = conf
 
-    val conf2 = if (cmd.operations.nonEmpty) {
-      conf.withValue(RUN_TABLES, ConfigValueFactory.fromIterable(cmd.operations.asJava))
-    } else conf
+    if (cmd.operations.nonEmpty)
+      accumulatedConfig = accumulatedConfig
+        .withValue(RUN_TABLES, ConfigValueFactory.fromIterable(cmd.operations.asJava))
 
-    val conf3 = cmd.rerunInfoDate match {
-      case Some(infoDate) =>
-        conf2
-          .withValue(IS_RERUN, ConfigValueFactory.fromAnyRef(true))
-          .withValue(CURRENT_DATE, ConfigValueFactory.fromAnyRef(dateFormatter.format(infoDate)))
-      case None           => conf2
-    }
+    for (infoDate <- cmd.rerunInfoDate)
+      accumulatedConfig = accumulatedConfig
+        .withValue(IS_RERUN, ConfigValueFactory.fromAnyRef(true))
+        .withValue(CURRENT_DATE, ConfigValueFactory.fromAnyRef(dateFormatter.format(infoDate)))
 
-    val conf4 = cmd.currentDate match {
-      case Some(date) => conf3
+    for (date <- cmd.currentDate)
+      accumulatedConfig = accumulatedConfig
         .withValue(CURRENT_DATE, ConfigValueFactory.fromAnyRef(dateFormatter.format(date)))
```

Comment:
This is great! So much better than numbered vals.

Reply:
Thank you for the fast review! 😄 I know that `var` is generally a no-no in Scala but thought it would maybe fit here.